### PR TITLE
Add get_gpu_instance() function and Organized the instance class codes.

### PR DESCRIPTION
**Pull request by whyb**
因为在实现AMD Vulkan Memory Allocator(VmaAllocator) 的时候，创建VmaAllocator的时候填写VmaAllocatorCreateInfo时候，需要提供与GPU设备相关的对象：
```cpp
VmaAllocatorCreateInfo allocatorInfo = {};
allocatorInfo.physicalDevice = vkdev->info.physical_device();
allocatorInfo.device = vkdev->vkdevice();
allocatorInfo.instance = get_gpu_instance(); //这个地方需要在allocator.cpp中去获取gpu.cpp中的VkInstance静态变量。
allocatorInfo.vulkanApiVersion = GetVulkanApiVersion();
```

故此pull request是将原本的 `gpu.cpp` 中的VkInstance暴露出来给外部使用，因为instance有create、destroy，顺便也应该对应有个get，就放在一起导出了哈。

改动后的的代码：

gpu.h
```cpp
namespace ncnn {
NCNN_EXPORT int create_gpu_instance();
NCNN_EXPORT VkInstance get_gpu_instance();
NCNN_EXPORT void destroy_gpu_instance();
```

gpu.cpp
```cpp
namespace ncnn {
static Mutex g_instance_lock;
class __ncnn_vulkan_instance_holder
{
public:
    __ncnn_vulkan_instance_holder()
    {
        instance = 0;
#if ENABLE_VALIDATION_LAYER
        callback = 0;
#endif
    }
    ~__ncnn_vulkan_instance_holder()
    {
        destroy_gpu_instance();
    }

    operator VkInstance()
    {
        return instance;
    }

    VkInstance instance;
#if ENABLE_VALIDATION_LAYER
    VkDebugUtilsMessengerEXT callback;
#endif
};
static __ncnn_vulkan_instance_holder g_instance;

...
int create_gpu_instance()
{
...
}

VkInstance get_gpu_instance()
{
    return (VkInstance)g_instance;
}

void destroy_gpu_instance()
{
...
}

```

具体详见 Files changed

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=4630) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=4630) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/ncnn?pullRequest=4630) it.</sub>

File: src/gpu.h [hunk 1]
```diff
@@ -28,9 +28,35 @@
...
+
+// Get global VkInstance variable
+// Must be called after create_gpu_instance() and before destroy_gpu_instance()
+NCNN_EXPORT VkInstance get_gpu_instance();
+
+// Destroy VkInstance object and free the memory of the associated object
+// Usually called in the destructor of the main program exit
 NCNN_EXPORT void destroy_gpu_instance();
 
+class __ncnn_vulkan_instance_holder
```
Comment (nihui) [inline src/gpu.h:46]: do not expose this internal class to public header

**Review by nihui — CHANGES_REQUESTED** [review]

File: src/gpu.h (hunk 1 above)
Comment (whyb) [reply src/gpu.h]: https://github.com/Tencent/ncnn/pull/4630/commits/e12c023df42f93b4dc333c93e755df72b1455468
Resolved.

**Comment by nihui** [comment]
Thanks for your contribution !

**Pull request merged**